__tests__/main.test.ts

```diff
@@ -243,3 +251,28 @@ test("Download file from release identified by ID", async () => {
   const result = await downloader.download(downloadSettings)
   expect(result.length).toBe(1)
 }, 10000)
+
+test("Download all archive files from public repo", async () => {
+  const extract_assats = true
+  const downloadSettings: IReleaseDownloadSettings = {
+    sourceRepoPath: "robinraju/probable-potato",
+    isLatest: true,
+    tag: "",
+    id: "",
+    fileName: "*.zip",
+    tarBall: false,
+    zipBall: false,
+    outFilePath: outputFilePath
+  }
+  const result = await downloader.download(downloadSettings)
+  if (extract_assats) {
```

Comment:
Related to my other comment about this input, we can avoid invoking `extract(asset, downloadSettings.outFilePath)` from this test if an `extractAssets` field is part of `IReleaseDownloadSettings`.
You just need to check the existence of the extracted file instead, as it will get downloaded based on the download settings.

Reply:
Thanks for the awesome feedback and your good eye on the input parsing bug!

I already fixed the other things but I'm not 100% sure what you mean with this comment. I added a field to the 'IReleaseDownloadSettings' interface and use it for checking now. But if you want to complacently avoid the extra invoction of `extract(asset, downloadSettings.outFilePath)` in the test file or even in the main file i would have to move it into the release-downloader.ts file. 

I thought about it in the beginning but I was not sure if you were okay with it, especially since the file is called  _downloader_ and not something like _worker_ or _extractor_.

Reply:
Please ignore that comment. Its better to keep download and extract separately.